python/pyarrow/_parquet.pyx

```diff
@@ -1278,11 +1297,43 @@ cdef shared_ptr[WriterProperties] _create_writer_properties(
 
     if isinstance(use_byte_stream_split, bool):
         if use_byte_stream_split:
-            props.encoding(ParquetEncoding_BYTE_STREAM_SPLIT)
+            if column_encoding is not None:
+                raise ValueError(
+                    "'column_encoding' is not None")
```

Comment:
This is a rather confusing error message. Perhaps explain the user what the conflicting options are?

Reply:
The reason behind all these conflicts is that `use_byte_stream_split` encoding is same as `BYTE_STREAM_SPLIT` column encoding option. To catch situations when both are specified, I added this checks.

I could change the error msg to "`use_byte_stream_split` can not be passed together with `column_encoding`" or maybe remove `use_byte_stream_split` altogether?

Reply:
Yes, my suggestion is to reword the error message.

Reply:
Sry, misunderstood a bit there. Will do.